### PR TITLE
unite titlefonts

### PR DESCRIPTION
**Pull request by BeastyBlacksmith**
I am willing to take a stab at https://github.com/JuliaPlots/Plots.jl/issues/302. The question is: do we want to make this a breaking change?
E.g. remove `titlefontcolor`, `titlefontsize`, etc. and let those all be `PlotText` objects.
Implementing `:match` could be tricky.

Or non-breaking: Accept those arguments but remove the fields from the `Plot` object and construct the `PlotText` objects internally.
The issue here is that certain titles like `plot_title` or https://github.com/JuliaPlots/Plots.jl/issues/463 don't have the `*fontsize` keywords so they would need to get added for consistency.

Related: https://github.com/JuliaPlots/Plots.jl/issues/1180

### CODE REVIEW

**Comment by daschw** [comment]
The first option would probably be nicer for internal development, BUT: If you only allow `PlotText` objects, it won't be possible to specify the title from within a recipe without depending on Plots anymore. So I'd rather go for the non-breaking change.

**Comment by BeastyBlacksmith** [comment]
Good point

File: src/args.jl [hunk 1]
```diff
@@ -288,13 +288,29 @@ const _series_defaults = KW(
...
-    :window_title                 => "Plots.jl",
+    :window_title                => "Plots.jl",
+    :window_titlefontsize        => 14,
+    :window_title_location       => :center,           # also :left or :right
+    :window_fontfamily           => :match,
+    :window_titlefontfamily      => :match,
+    :window_titlefonthalign      => :hcenter,
+    :window_titlefontvalign      => :vcenter,
+    :window_titlefontrotation    => 0.0,
+    :window_titlefontcolor       => :match,
```
Comment (daschw) [inline src/args.jl]: Does the window title really have all these attributes? Isn't it just the text appearing in the gui window's title bar? In that case the font, etc. would depend on the OS, right? Or am I missing another use case for `window_title`?

Comment (BeastyBlacksmith) [reply src/args.jl]: You are  right. Thanks for pointing out. I didn't really think here. I will delete these and improve the description of `window_title` instead.

**Comment by BeastyBlacksmith** [comment]
I would say we don't need to actually create the `PlotText` objects, since we have the `*title` fields and the `*font` accessor-functions.
Only question that I have is how do we allow for string and `PlotText` input to the `*title` fields?

**Comment by BeastyBlacksmith** [comment]
> The first option would probably be nicer for internal development, BUT: If you only allow `PlotText` objects, it won't be possible to specify the title from within a recipe without depending on Plots anymore. So I'd rather go for the non-breaking change.

This wouldn't be an issue, if `PlotText` would live in `RecipesBase`, right? Is there any downside to this? @daschw

**Comment by daschw** [comment]
We want to keep RecipesBase as minimal as possible to reduce the barrier for other packages to take the depency. The same argument as for `PlotText` could be made for `Stroke`, `Brush`, `Font` (already required by `PlotText`) `Layout` and probably more. Also the idea of the recipe system is that other plotting libraries could hook into it as well. While currently only Plots uses it, the aim is that Makie can also consume RecipesBase recipes (I think).
So I'm not sure, if in practice there would be a big downside. But until now our strategy has been not to move too much from Plots to RecipesBase. What do you think @mkborregaard ?

**Comment by mkborregaard** [comment]
Agreed, our policy has been not to move concrete types. 
BUT, we have a plottext constructor that takes tuples, and process them like a magic argument. I've used that e.g. here: https://github.com/richardreeve/Phylo.jl/blob/master/src/plot.jl#L103-L104
Could that approach not resolve it?

**Comment by BeastyBlacksmith** [comment]
So the plan would be to have all text keywords like titles to use the `plottext` constructor to construct `PlotText` objects and then we can get rid of all the font related keywords, if I got that right.

**Comment by BeastyBlacksmith** [comment]
added the magic arguments, so now you can construct them like the other fonts, too.
I think we can spare the `PlotText` refacturing for 2.0 and keep it as is?
Still has to be implemented from the backend side and maybe add aliases.
I can add an implementation for the pgfplotsx backend either on this PR or in a seperate PR.

**Comment by BeastyBlacksmith** [comment]
I think this is fine now. @daschw, @mkborregaard  any objections?

**Pull request merged**